jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -0,0 +1,511 @@
+                                         'bg_results_label_default',
+                                         'bg_results_label_auto',
+                                         'bg_results_label_invalid_msg',
+                                         'bg_results_label_overwrite',
+                                         'bg_add_to_viewer_items',
+                                         'bg_add_to_viewer_selected')
+        self.bg_add_results.viewer.filters = ['is_spectrum_2d_viewer']
+        self.bg_results_label_default = 'background'
+
+        self.bg_sub_add_results = AddResults(self, 'bg_sub_results_label',
+                                             'bg_sub_results_label_default',
+                                             'bg_sub_results_label_auto',
+                                             'bg_sub_results_label_invalid_msg',
+                                             'bg_sub_results_label_overwrite',
+                                             'bg_sub_add_to_viewer_items',
+                                             'bg_sub_add_to_viewer_selected')
+        self.bg_sub_add_results.viewer.filters = ['is_spectrum_2d_viewer']
+        self.bg_sub_results_label_default = 'background-subtracted'
+
+        # EXTRACT
+        self.ext_dataset = DatasetSelect(self,
+                                         'ext_dataset_items',
+                                         'ext_dataset_selected',
+                                         default_text='From Plugin',
+                                         filters=['layer_in_spectrum_2d_viewer', 'not_trace'])
+
+        self.ext_add_results = AddResults(self, 'ext_results_label',
+                                          'ext_results_label_default',
+                                          'ext_results_label_auto',
+                                          'ext_results_label_invalid_msg',
+                                          'ext_results_label_overwrite',
+                                          'ext_add_to_viewer_items',
+                                          'ext_add_to_viewer_selected')
+        self.ext_add_results.viewer.filters = ['is_spectrum_viewer']
+        # NOTE: defaults to overwriting original spectrum
+        self.ext_add_results.label_whitelist_overwrite = ['Spectrum 1D']
+        self.ext_results_label_default = 'Spectrum 1D'
+
+    @observe('trace_dataset_selected')
+    def _trace_dataset_selected(self, msg=None):
+        if not hasattr(self, 'trace_dataset'):
+            # happens when first initializing plugin outside of tray
+            return
+
+        width = self.trace_dataset.selected_obj.shape[0]
+        # estimate the pixel number by taking the median of the brightest pixel index in each column
+        brightest_pixel = int(np.median(np.argmax(self.trace_dataset.selected_obj.flux, axis=0)))
+        # default width will be 10% of cross-dispersion "height"
+        default_width = int(np.ceil(width / 10))
```

Comment:
this might warrant some discussion - as it dictates the default extraction inputs when a 2D spectrum is loaded without a 1D spectrum

Reply:
I think this is fine for now, this isn't a robust pipeline and we throw a warning to not trust this too much.